RecordCollection/Controllers/AlbumsController.cs

```diff
@@ -15,8 +19,39 @@ public AlbumsController(RecordCollectionContext context)
         public IActionResult Index()
         {
             var albums = _context.Albums.ToList();
-            
             return View(albums);
         }
+
+        [Route("Albums/{id:int}")]
+        public async Task<IActionResult> Show(int? id)
+        {
+            if (id == null)
```

Comment:
We will be null checking a lot of values across our app, could we extract this into its own method? For the method I would want it take in multiple different types and check if they are null then return `NotFound()`... Possibly using Generics?